src/components/dbCalls.js

```diff
@@ -0,0 +1,435 @@
+    })
+    .catch((error) => {
+        console.error('Error adding player: ', error);
+        return createAlert('error', 'Error adding player:', 'check console', 1500);
+    });
+}
+
+//removes player from database
+const removePlayerForRoom = async (player, createAlert, roomID) => {
+    const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+    const playerQuery = query(playerCollectionRef, where('name', '==', player));
+    const playerSnapshot = await getDocs(playerQuery); 
+    //returns error alert if player not found
+    if (playerSnapshot.empty) return createAlert('error', 'Error', 'player does not exist', 1500);
+    try {
+        const docRef = playerSnapshot.docs[0].ref;
+        await deleteDoc(docRef);
+    }
+    catch (error) {
+        console.error('Error removing player: ', error);
+        createAlert('error', 'Error deleting player', 'check console', 1500);
+    }
+}
+
+//dupates assassins of player in database
+const updateAssassinsForPlayer = async (player, assassins, createAlert, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { assassins: assassins });
+    }
+    catch (error) {
+        console.error('Error updating player assassins: ', error);
+        createAlert('error', 'Error updating player assassins', 'check console', 1500);
+    }
+}
+
+//updates targets of player in database
+const updateTaregtsForPlayer = async (player, targets, createAlert, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { targets: targets });
+    }
+    catch (error) {
+        console.error('Error updating player targets: ', error);
+        createAlert('error', 'Error updating player targets', 'check console', 1500);
+    }
+}
+
+//fetches player's assassins
+const fetchAssassinsForPlayer = async (player, createAlert, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerAssassins = playerSnapshot.docs[0].data().assassins;
+        return playerAssassins;
+    }
+    catch (error) {
+        console.error('Error fetching player assassins: ', error);
+        createAlert('error', 'Error fetching player assassins', 'check console', 1500);
+    }
+}
+
+const fetchReferenceForTask = async (taskID, createAlert, roomID) => {
```

Comment:
Similar to comment above, let's not worry about creating an alert. If an error happens, throw an error, and catch it wherever this was called. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Control_flow_and_error_handling#exception_handling_statements